File: app-miyou/src/main/java/com/miyou/app/infrastructure/mission/repository/MissionMongoRepository.java
package com.miyou.app.infrastructure.mission.repository;

import org.springframework.data.mongodb.repository.ReactiveMongoRepository;

import com.miyou.app.infrastructure.mission.document.MissionDocument;
import reactor.core.publisher.Flux;

public interface MissionMongoRepository
	extends ReactiveMongoRepository<MissionDocument, String> {

	Flux<MissionDocument> findByType(String type);
}

File: app-miyou/src/main/kotlin/com/miyou/app/infrastructure/mission/repository/MissionMongoRepository.kt
package com.miyou.app.infrastructure.mission.repository

import com.miyou.app.infrastructure.mission.document.MissionDocument
import org.springframework.data.mongodb.repository.ReactiveMongoRepository
import reactor.core.publisher.Flux

interface MissionMongoRepository : ReactiveMongoRepository<MissionDocument, String> {
    fun findByType(type: String): Flux<MissionDocument>
}
